lib/core/src/chain_swap.rs

```diff
@@ -367,6 +367,29 @@ impl ChainSwapHandler {
             | ChainSwapStates::TransactionLockupFailed
             | ChainSwapStates::TransactionRefunded
             | ChainSwapStates::SwapExpired => {
+                // Zero-amount Receive Chain Swaps also get to TransactionLockupFailed when user locks up funds
+                let is_zero_amount = swap.payer_amount_sat == 0;
+                if matches!(swap_state, ChainSwapStates::TransactionLockupFailed) && is_zero_amount
+                {
+                    if let Err(e) = self
+                        .swapper
+                        .get_zero_amount_chain_swap_quote(&swap.id)
+                        .map(|quote| quote.to_sat())
+                        .and_then(|quote| {
+                            info!("Got quote of {quote} sat for swap {}", &swap.id);
+
+                            self.persister.update_swap_payer_amount(&swap.id, quote)?;
```

Comment:
It looks like the quote here is the [server lockup amount](https://github.com/BoltzExchange/boltz-backend/commit/b93f0e8a0ef61c47b2160600d588294c75ae9d58#diff-05ebf780e4d6d2c474c4a231242487fcc27906cb3672d657a20267b91803fa89R47). If so, I think there needs to be some recalculations done. The receiver amount (quote - claim fees), fees (server fees + boltz fees + claim fees) and payer amount (receiver amount + claim fees + server fees + boltz fees) need calculating.